Classes/Utility/ClassCacheManager.php

```diff
@@ -162,12 +169,20 @@ protected function parseSingleFile($filePath, $baseClass = false): string
         // unset the constructor and save it's lines
         if (isset($classParserInformation['functions']['__construct'])) {
             $constructorInfo = $classParserInformation['functions']['__construct'];
-            for ($i = $constructorInfo['start'] - $offsetForInnerPart; $i < $constructorInfo['end'] - $offsetForInnerPart; $i++) {
-                if (trim($innerPart[$i]) === '{') {
+            $docLineCount = substr_count($constructorInfo['doc'], "\n") + 1;
+            if($baseClass) $this->constructorLines['doc'] = $constructorInfo['doc'];
+            $unsetLine = true;
+            for ($i = ($constructorInfo['start']  - $docLineCount - 1) - ($offsetForInnerPart); $i < $constructorInfo['end'] - $offsetForInnerPart; $i++) {
+                if ($unsetLine) {
+                    if(trim($innerPart[$i]) === ') {' || trim($innerPart[$i]) === '{') {
+                        $unsetLine = false;
+                    } elseif($baseClass && trim($innerPart[$i]) !== ')' && $i > ($constructorInfo['start'] - 1)) {
+                        $this->constructorLines['parameters'][] = LF . $innerPart[$i];
```

Comment:
@stigfaerch, so it's not possible to add other parameters to the constructor? Like EXT:eventnews's `LocationRepository $locationRepository` parameter to the NewsController? https://github.com/georgringer/eventnews/pull/125/files#diff-4cff6710bf1e915a96f471e0a75cbbfe71dc07cf2eebabdfe07cd54878fbce14R38

Reply:
Should work now.